velox/exec/tests/utils/PrestoQueryRunner.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <folly/SocketAddress.h>
+#include <folly/io/async/EventBaseThread.h>
+#include "velox/common/memory/Memory.h"
+#include "velox/exec/tests/utils/ReferenceQueryRunner.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::exec::test {
+class PrestoQueryRunner : public velox::exec::test::ReferenceQueryRunner {
+ public:
+  /// @param coordinatorUri Presto REST API endpoint, e.g. http://127.0.0.1:8080
+  /// @param user Username to use in X-Presto-User header.
+  PrestoQueryRunner(
+      std::string coordinatorUri,
+      std::string user,
+      const int32_t timeout = 10'000);
```

Comment:
- Drop const.
- Document this parameter.
- Perhaps, add units to the name for clarity: timeoutMillis or use an [std::chrono](https://en.cppreference.com/w/cpp/header/chrono) type.

Why 10'000? Is this 10 seconds?

Why does the test need to override this to 20'000?

Reply:
I've removed the const, documented this parameter, and used std::chrono type.

>Why 10'000? Is this 10 seconds?

I followed the default value used in PrestoDB. I just changed it to 1000ms.

>Why does the test need to override this to 20'000?

Removed.

Reply:
Thanks.

Reply:
@mbasmanova Could we add the document file in the fellow-up PR? I found that I need to make code changes to show how to use `PrestoQueryRunner` in Velox. I want to add a follow-up PR to make the runner configurable through the `gflags` along with the document on how to use PrestoQueryRunner. What do you think?

Reply:
> Could we add the document file in the fellow-up PR? 

Sure.

> I want to add a follow-up PR to make the runner configurable through the gflags

Wondering why would we want to do that.

Reply:
>Wondering why would we want to do that.

I found that I need to make code changes to show how to use PrestoQueryRunner in Velox, say `velox_aggregation_fuzzer_test`, when I am trying to write the document of how to use PrestoQueryRunner, something like the following. With the follow-up, we could control the creation of the query runner instance to switch runner only through program parameters.

```C++
diff --git a/velox/exec/tests/AggregationFuzzerTest.cpp b/velox/exec/tests/AggregationFuzzerTest.cpp
--- a/velox/exec/tests/AggregationFuzzerTest.cpp
+++ b/velox/exec/tests/AggregationFuzzerTest.cpp

+#include "velox/exec/tests/utils/PrestoQueryRunner.h"
-  auto duckQueryRunner =

+  auto queryRunner =
+      std::make_unique<facebook::velox::exec::test::PrestoQueryRunner>("http://127.0.0.1:8080", "hive");

@@ -343,5 +337,5 @@ int main(int argc, char** argv) {
       facebook::velox::exec::test::getCustomInputGenerators();
   options.timestampPrecision =
       facebook::velox::VectorFuzzer::Options::TimestampPrecision::kMilliSeconds;
-  return Runner::run(initialSeed, std::move(duckQueryRunner), options);
+  return Runner::run(initialSeed, std::move(queryRunner), options);
```

Reply:
>Once Presto query runner works, we want to always use that and remove Duck query runner.

Got, it. I will highlight this in the follow-up.